doc/reference/reference_lua/box_watchers/watch.rst

```diff
@@ -0,0 +1,43 @@
+.. _box-watch:
+
+================================================================================
+box.watch()
+================================================================================
+
+..  function:: box.watch(key, func)
+
+    Subscribe to events broadcast by a local host.
+    It is possible to register more than one watcher for the same key.
+
+    :param string key: a key name of an event to subscribe to
+    :param function func: a callback to invoke when the key value is updated
+
+    :return: a watcher handle that can be used to unregister the watcher
+
+    A watcher callback is first invoked unconditionally after the watcher registration.
+    Subsequent invocations are triggered by :doc:`box.broadcast() </reference/reference_lua/box_watchers/broadcast>`
+    called on the remote host.
+    A watcher callback is passed the key for which it was registered and the current key value.
+    A watcher callback is always invoked in a new fiber so it is okay to yield in it.
+    A watcher callback is never executed in parallel with itself.
+    If the key is updated while the watcher callback is running, the callback will be invoked again with the new
+    value as soon as it returns.
```

Comment:
I think we should also mention what happens if a watcher is registered on a key that has never been broadcasted so far.

Reply:
fixed